src/pages/index.js

```diff
@@ -87,25 +119,92 @@ const addFormValidator = new FormValidator(
   cardForm
 );
 
+const avatarFormValidator = new FormValidator(
+  constants.validationSettings,
+  avatarForm
+);
+
 const profileInfo = new UserInfo({
   name: ".profile__name",
   description: ".profile__description",
 });
 
-const editProfilePopup = new PopupWithForm("#edit-popup", handleSaveProfile);
-const newCardPopup = new PopupWithForm("#new-card-popup", handleAddCard);
+const editProfilePopup = new PopupWithForm("#edit-popup", handleUpdateProfile);
+
+const updateAvatarPopup = new PopupWithForm(
+  "#avatar-popup",
+  handleUpdateAvatar
+);
+
 const imagePopup = new PopupWithImage("#preview-image-popup");
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
 // ! ||--------------------------------------------------------------------------------||
 // ! ||                                Gallery Creation                                ||
 // ! ||--------------------------------------------------------------------------------||
 
-const cardGallery = new Section(
-  { items: constants.initialCards, renderer: renderCard },
-  ".gallery__cards"
-);
-
-cardGallery.renderItems();
+api.getInitialCards().then((cards) => {
+  const cardGallery = new Section(
+    { items: cards, renderer: renderCard },
+    ".gallery__cards"
+  );
+
+  function renderCard(cardData) {
+    const card = new Card(
+      {
+        cardData,
+
+        handleImageClick,
+        handleDeleteButton,
+        handleCardLike,
+        handleCardUnlike,
+      },
+      cardTemplate
+    );
+
+    function handleDeleteButton() {
+      console.log(document.querySelector(`.modal_type_delete`));
+      deletePopup.open();
+    }
+
+    function handleCardLike(id) {
+      api.likeCard(id);
+    }
+
+    function handleCardUnlike(id) {
+      api.unlikeCard(id);
+    }
+
+    const handleDeleteSubmit = () => {
+      let id = card.getId();
+      api.removeCard(id);
+      card.removeItem();
+      // For some reason, when this function is called consecutively,
+      // it saves the card's ID from each previous time it was called,
+      // so it tries to delete those IDs as well.
+    };
+
+    const deletePopup = new PopupDelete("#delete-popup", handleDeleteSubmit);
```

Comment:
Needs correction: There should be only one global instance of the confirmation popup. The confirmation handler should be declared when you click the trash button, you should pass it to the delete popup instance via action setter

Example of confirm popup methods: 

```js
constructor(popupSelector) {
  super(popupSelector);
  this._form = this._modal.querySelector(".modal__form");
}

setAction(action) { 
  this._handleFormSubmit = action; 
} 

this._form.addEventListener("submit", (evt) => { 
  evt.preventDefault(); 
  this._handleFormSubmit(); 
})
```

Example of card deletion:

```js
const confirmDeletePopup = new PopupConfirm("#confirm-delete-popup"); // note, that it doesn't accept handlers

function handleTrashClick(cardInstance) {
  confirmDeletePopup.open();
  confirmDeletePopup.setAction(() => {
   // delete the card using Card instance
  });
}
```